src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -934,7 +935,12 @@ class spell_mage_pyroblast_clearcasting_driver : public AuraScript
         if (!caster->HasAura(SPELL_MAGE_HEATING_UP) && !caster->HasAura(SPELL_MAGE_HOT_STREAK))
         {
             caster->CastSpell(caster, SPELL_MAGE_HEATING_UP, true);
+
             procCheck = true;
+
+            if (AuraEffect* burn = caster->GetAuraEffect(SPELL_MAGE_CONTROLLED_BURN, EFFECT_0))
+                if (roll_chance_i(burn->GetAmount()))
```

Comment:
Shouldn't that be if (!roll_chance_i(burn->GetAmount())) ?

Reply:
it should have 20% no?